Fix clippy lints

simonsan opened the pull request:
API-Breaking change:
Rename `to_string_lang` -> `get_string_from_lang` due to taking `&self` which is not conventional for `to_*`-functions

r? @the-eater @goto-bus-stop

the-eater commented on crates/genie-dat/src/tech_tree.rs:
```diff
@@ -194,6 +194,8 @@ impl TryFrom<i32> for TechTreeDependencyType {
 }
 
+// FIXME: implementation of `From` is preferred since it gives you `Into<_>`
+// for free where the reverse isn't true
```
Does something break by replacing this with a `From`?

the-eater commented on crates/genie-hki/src/lib.rs:
```diff
@@ -917,7 +917,7 @@ impl HotkeyGroup {
     /// Returns a string representation of this hotkey group, using the strings
     /// from `lang_file` and the group name string key `sk`.
-    pub fn to_string_lang(&self, lang_file: &LangFile, sk: &StringKey) -> String {
+    pub fn return_string_from_lang(&self, lang_file: &LangFile, sk: &StringKey) -> String {
```
`get_string_from_lang`?

the-eater reviewed (commented):
Looks good!

could you maybe split the clippy fixes and package updates into separate commits?

simonsan committed:
Clippy fixes

API-breaking change:
Rename `to_string_lang` -> `get_string_from_lang` due to taking `&self` which is not conventional for `to_*`-methods

simonsan commented:
> could you maybe split the clippy fixes and package updates into separate commits?

Done

the-eater commented on crates/genie-drs/src/lib.rs:
```diff
@@ -171,9 +171,9 @@ impl DRSHeader {
-        let mut password = [0 as u8; 12];
+        let mut banner_msg = [0_u8; 40];
+        let mut version = [0_u8; 4];
+        let mut password = [0_u8; 12];
```
is `0_u8` a personal preference over `0u8`?

simonsan replied:
imho it's better visible in the source code with an underscore. and let's number part and type easily being distinguished from one another. not sure if only personal preference though.
